### PR TITLE
Additional config

### PR DESCRIPTION
**Pull request by ghost**
Added dockerignore, Compose file for Prometheus, and changed monitoring logic to allow for non-IP Prometheus addresses

### CODE REVIEW

**Comment by ghost** [comment]
I'm actually not sure what the Prometheus config should be to check it, and switching to the config.toml lost a crashing on a nonsense webhook.

If possible it'd be nice to attempt Prometheus connection outside of Tokio to crash early instead of crash inside it then keep the main process running (IMO)

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    env_file:
+      - .env
+    environment:
+      - HIGHLIGHTS_CONFIG=/run/secrets/config.toml
+    networks:
+      - highlights-net
+    secrets:
+      - config.toml
+    volumes:
+      # This should match the HIGHLIGHTS_DATA_DIR environmental variable
```
Comment (ThatsNoMoon) [inline docker-compose.yml]: This should probably say "configured database path". After https://github.com/ThatsNoMoon/highlights/pull/37 the environment variable is `HIGHLIGHTS_DATABASE.PATH` now and would probably be configured as
```toml
[database]
path = "..."
```
in most cases anyway.

File: src/monitoring.rs [hunk 2]
```diff
@@ -200,9 +203,22 @@ fn avg_metrics(metric_families: Vec<MetricFamily>) -> Option<f64> {
...
 pub fn init() {
-	if let Some(addr) = settings().logging.prometheus {
-		ENABLED.set(true).unwrap();
-		tokio::spawn(prometheus_server(addr));
+	if let Some(addr) = &settings().logging.prometheus {
+		match addr.to_socket_addrs() {
+			Ok(mut sock_addr) => {
+				ENABLED.set(true).unwrap();
+				tokio::spawn(prometheus_server(
+					sock_addr.next().expect("No socket address provided"),
```
Comment (ThatsNoMoon) [inline src/monitoring.rs]: Can this ever fail? I don't think so from what I can tell so maybe something like `"String produced no socket addresses"` would be better.

**Comment by ThatsNoMoon** [comment]
Since https://github.com/ThatsNoMoon/highlights/commit/54726d0f06a3749373268f67cc79508ef8d5e3d4, Prometheus isn't being used anymore, so I'll (finally) close this.

**Pull request closed without merging**